blockchain/blockdao/blockdao.go

```diff
@@ -174,9 +177,53 @@ func (dao *blockDAO) Start(ctx context.Context) error {
 			return errors.Wrap(err, "failed to write initial value for top height")
 		}
 	}
-	return dao.initStores()
+	err = dao.initStores()
+	if err != nil {
+		return err
+	}
+	return dao.adjust()
+}
+func (dao *blockDAO) adjust() error {
+	topHeight, err := dao.getTipHeight()
+	if err != nil {
+		return err
+	}
+	topDB, index, err := dao.getTopDB(topHeight)
+	if err != nil {
+		return err
+	}
+	if index == 0 {
```

Comment:
index == 0 shouldn't be a corner case. It doesn't hurt to do the checking. In other words, we suppose to see:
chain.db, chain_{height1}.db, chain_{height2}.db, ...

Storing blocks in chain.db is not a good idea, which is very confusing.

Reply:
If you think storing blocks in chain.db is not good,I'll try to change that

Reply:
Make the change pls.